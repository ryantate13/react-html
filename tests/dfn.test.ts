import {dfn} from '../src';

describe('dfn', () => {
    it('renders with children', () => {
        expect(dfn({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(dfn({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(dfn()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(dfn({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(dfn({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(dfn({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(dfn({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(dfn({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(dfn({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(dfn({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(dfn({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(dfn({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(dfn({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(dfn({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(dfn({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(dfn({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(dfn({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(dfn({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(dfn({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(dfn({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(dfn({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(dfn({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(dfn({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(dfn({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(dfn({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(dfn({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(dfn({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(dfn({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(dfn({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(dfn({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(dfn({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(dfn({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(dfn({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(dfn({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(dfn({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(dfn({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(dfn({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(dfn({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(dfn({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(dfn({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(dfn({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(dfn({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(dfn({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(dfn({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(dfn({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(dfn({translate: "no"})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(dfn({title: "TEST"})).toMatchSnapshot();
    });
});