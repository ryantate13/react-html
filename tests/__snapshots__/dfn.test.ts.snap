// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dfn renders arrays of children 1`] = `
<dfn>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</dfn>
`;

exports[`dfn renders with accessKey="a,b,c,d" 1`] = `
<dfn
  accessKey="a,b,c,d"
/>
`;

exports[`dfn renders with autoCapitalize="characters" 1`] = `
<dfn
  autoCapitalize="characters"
/>
`;

exports[`dfn renders with autoCapitalize="none" 1`] = `
<dfn
  autoCapitalize="none"
/>
`;

exports[`dfn renders with autoCapitalize="off" 1`] = `
<dfn
  autoCapitalize="off"
/>
`;

exports[`dfn renders with autoCapitalize="on" 1`] = `
<dfn
  autoCapitalize="on"
/>
`;

exports[`dfn renders with autoCapitalize="sentences" 1`] = `
<dfn
  autoCapitalize="sentences"
/>
`;

exports[`dfn renders with autoCapitalize="words" 1`] = `
<dfn
  autoCapitalize="words"
/>
`;

exports[`dfn renders with children 1`] = `
<dfn>
  TEST
</dfn>
`;

exports[`dfn renders with className="foo bar baz" 1`] = `
<dfn
  className="foo bar baz"
/>
`;

exports[`dfn renders with contentEditable="false" 1`] = `
<dfn
  contentEditable="false"
/>
`;

exports[`dfn renders with contentEditable="true" 1`] = `
<dfn
  contentEditable="true"
/>
`;

exports[`dfn renders with dir="auto" 1`] = `
<dfn
  dir="auto"
/>
`;

exports[`dfn renders with dir="ltr" 1`] = `
<dfn
  dir="ltr"
/>
`;

exports[`dfn renders with dir="rtl" 1`] = `
<dfn
  dir="rtl"
/>
`;

exports[`dfn renders with draggable="false" 1`] = `
<dfn
  draggable="false"
/>
`;

exports[`dfn renders with draggable="true" 1`] = `
<dfn
  draggable="true"
/>
`;

exports[`dfn renders with empty attributes 1`] = `<dfn />`;

exports[`dfn renders with hidden=false 1`] = `
<dfn
  hidden={false}
/>
`;

exports[`dfn renders with hidden=true 1`] = `
<dfn
  hidden={true}
/>
`;

exports[`dfn renders with id="TEST" 1`] = `
<dfn
  id="TEST"
/>
`;

exports[`dfn renders with inputMode="decimal" 1`] = `
<dfn
  inputMode="decimal"
/>
`;

exports[`dfn renders with inputMode="email" 1`] = `
<dfn
  inputMode="email"
/>
`;

exports[`dfn renders with inputMode="none" 1`] = `
<dfn
  inputMode="none"
/>
`;

exports[`dfn renders with inputMode="numeric" 1`] = `
<dfn
  inputMode="numeric"
/>
`;

exports[`dfn renders with inputMode="search" 1`] = `
<dfn
  inputMode="search"
/>
`;

exports[`dfn renders with inputMode="tel" 1`] = `
<dfn
  inputMode="tel"
/>
`;

exports[`dfn renders with inputMode="text" 1`] = `
<dfn
  inputMode="text"
/>
`;

exports[`dfn renders with inputMode="url" 1`] = `
<dfn
  inputMode="url"
/>
`;

exports[`dfn renders with is="foo" 1`] = `
<dfn
  is="foo"
/>
`;

exports[`dfn renders with itemID="https://google.com/" 1`] = `
<dfn
  itemID="https://google.com/"
/>
`;

exports[`dfn renders with itemProp="https://google.com/" 1`] = `
<dfn
  itemProp="https://google.com/"
/>
`;

exports[`dfn renders with itemRef="foo bar" 1`] = `
<dfn
  itemRef="foo bar"
/>
`;

exports[`dfn renders with itemScope=false 1`] = `
<dfn
  itemScope={false}
/>
`;

exports[`dfn renders with itemScope=true 1`] = `
<dfn
  itemScope={true}
/>
`;

exports[`dfn renders with itemType="https://google.com/" 1`] = `
<dfn
  itemType="https://google.com/"
/>
`;

exports[`dfn renders with lang="" 1`] = `
<dfn
  lang=""
/>
`;

exports[`dfn renders with lang="es-419" 1`] = `
<dfn
  lang="es-419"
/>
`;

exports[`dfn renders with slot="TEST" 1`] = `
<dfn
  slot="TEST"
/>
`;

exports[`dfn renders with spellCheck="false" 1`] = `
<dfn
  spellCheck="false"
/>
`;

exports[`dfn renders with spellCheck="true" 1`] = `
<dfn
  spellCheck="true"
/>
`;

exports[`dfn renders with style={"color":"#000000"} 1`] = `
<dfn
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`dfn renders with tabIndex=1 1`] = `
<dfn
  tabIndex={1}
/>
`;

exports[`dfn renders with title="TEST" 1`] = `
<dfn
  title="TEST"
/>
`;

exports[`dfn renders with title="TEST" 2`] = `
<dfn
  title="TEST"
/>
`;

exports[`dfn renders with translate="no" 1`] = `
<dfn
  translate="no"
/>
`;

exports[`dfn renders with translate="yes" 1`] = `
<dfn
  translate="yes"
/>
`;

exports[`dfn renders without attributes 1`] = `<dfn />`;
